### DISCUSSION TITLE
Can you cache node_modules?

### DISCUSSION THREAD

**Discussion by nandorojo** [vercel/turborepo#666] [Help] [answered]
Assuming there are no new `patches` or changes to `yarn.lock`/`package.json`, would it be possible to cache the `node_modules` folder for Vercel builds? My build takes a long time at the `yarn install` step, and I'm not sure if Turbo is supposed to help with that or not.

**Solved:** https://github.com/vercel/turborepo/discussions/666#discussioncomment-2616918

**Comment by phelma** [comment]
I think turborepo stores its cache in `node_modules/.cache` so it might need some work, would save a lot of time though

**Reply by nandorojo** [reply]
just updated with the working answer!

**Reply by mcrowder65** [reply]
So are you just running, turbo run install?

**Comment by nandorojo** [comment, marked as answer]
turns out you can do it if you set this env variable on Vercel:

`ENABLE_ROOT_PATH_BUILD_CACHE=1`

https://twitter.com/jaredpalmer/status/1488954563533189124?s=21&t=Ns4h0N-c174RceaHSm_vhA

**Reply by sppatel** [reply]
Is this vercel specific - or can u do this with a custom remote cache?

**Reply by nandorojo** [reply]
not sure. probably vercel-specific
